### PR TITLE
PS3 enhancement attempt

### PR DESCRIPTION
**Pull request by Sami32**
This change is [Reviewable](https://reviewable.io/reviews/digitalmediaserver/digitalmediaserver/113)

### CODE REVIEW

**Comment by Sami32** [comment]
@MadokaAyukawa Thank you for having taken the time to report :+1:

It will be nice if you could test and report back if the removed codecs that was not in the PS3 Sony specifications was really necessary :-)

**Comment by Sami32** [comment]
Note:
To get the raw file and replace it into your DMS install you need to click on "Files changed" tab, then on the "View" button and then on the "RAW" button.

**Comment by Sami32** [comment]
https://raw.githubusercontent.com/DigitalMediaServer/DigitalMediaServer/0ba1c5629aa231de3c27071ea7ad83e8567ecd8e/src/main/external-resources/renderers/Sony-PlayStation3.conf

**Comment by MadokaAyukawa** [comment]
I've started testing this file. I removed the "seekbytime" line as the version I'm using does not need that.

There needs to be a 4 in mpeg4asp in the avi line. I'm still new to GitHub so I'm not sure how to show you in the file. I don't want to mess it up.

I found that the following file is streamable, although DMS wants to tranascode it.

```
TRACE 2018-10-25 15:41:32.435 [HTTPv2 Request Worker 16] Matched format MPG to "C:\Users\ayato\Videos\[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4"
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] MediaInfo parsing results for "C:\Users\ayato\Videos\[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4":
  Format: "MPEG-4"              CodecID: "isom"               Duration: 0:29:40.845         Overall Bitrate Mode: VBR
  Overall Bitrate: 1467601      Cover: No                     FPS: 29.970
  - Video track ID: 1
    Format: "AVC"                 Profile: "High@L4.2"          CodecID: "avc1"               Duration: 0:29:40.845
    Bitrate: 1335144              BitRate Maximum: 2769472      Width: 640                    Height: 480
    Colorimetry: 4:2:0            Chroma: 4:2:0                 PAR: 1.000                    DAR: 4:3
    Scan Type: Progressive        FPS: 29.970                   FPS Orig: 29.970              Framerate Mode: CFR
    RefFrames: 4                  Bit Depth: 8
  - Audio track ID: 2
    Format: "AAC"                 Profile: "LC"                 CodecID: "mp4a-40-2"          Duration: 0:29:40.672
    BitRate Mode: VBR             Bitrate: 127912               BitRate Maximum: 155480       Channel(s): 2
    Samplerate: 48000
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Parsing results for file "[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4": Container: MP4, Size: 326696116, Video Bitrate: 1467601, Video Bitrate Mode: VARIABLE, Video Tracks: 1, Video Codec: h264, Duration: 0:29:40.845, Video Resolution: 640 x 480, Display Aspect Ratio: 4:3, Scan Type: Progressive, Framerate: 29.970, Original Framerate: 29.970, Framerate Mode: CFR, AVC Level: 4.2, Video Track Title from Metadata: Imported with GPAC 0.4.6-DEV (internal rev. 5), Audio Tracks: 1 [Audio Codec: AAC-LC, Bitrate: 127912, Bitrate Mode: VARIABLE, Channels: 2, Sample Frequency: 48000 Hz], Mime Type: video/mp4
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Adding new child "[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4" with class "RealFile"
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] This video does not have subtitles
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Getting player for resource "[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4"
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Returning compatible player "FFmpeg Video"
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Format "mp4" failed to match supported line f:avi|divx   v:mpeg4asp|mpeg4sp|divx|mjpeg|xvid  a:mp3|lpcm|mpa|ac3       m:video/x-divx      gmc:0
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Audio codec "aac-lc" failed to match support line f:mp4        v:h264                              a:ac3                    m:video/mp4
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Matched support line f:mp4        v:h264                              a:aac-lc                 m:video/mp4         n:2
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] File "[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4" will not be streamed because the H.264 level (4.2) is not supported.
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Format "mpegps" failed to match supported line f:avi|divx   v:mpeg4asp|mpeg4sp|divx|mjpeg|xvid  a:mp3|lpcm|mpa|ac3       m:video/x-divx      gmc:0
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Format "mpegps" failed to match supported line f:mp4        v:h264                              a:ac3                    m:video/mp4
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Format "mpegps" failed to match supported line f:mp4        v:h264                              a:aac-lc                 m:video/mp4         n:2
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Matched support line f:mpegps     v:mpeg1|mpeg2                       a:aac-lc|ac3|lpcm|mpa    m:video/mpeg
TRACE 2018-10-25 15:41:32.565 [HTTPv2 Request Worker 16] Final verdict: "[SkewedS]_Mechaike_No_Reaction_Drama_[B5361302].mp4" will be transcoded with player "FFmpeg Video" with mime type "video/mpeg"
```

**Comment by Nadahar** [comment]
The file is h264 level 4.2. I thought the PS3 was only level 4.1. Many renderers can still play some files with higher profiles, but there's no guarantee. The safest choice is probably to keep the limit at 4.1.

**Comment by Sami32** [comment]
@MadokaAyukawa Arrgh...sorry for the typo

Yes that is normal, as in the actual code the 4.2 level is not manageable.
So you could add the `H264Level41Limited = false` parameter into your configuration file but i didn't set it by default here, though i did in my fork, as it will give issue for users as soon as they will have a media with a video level > 4.2
Our PR related to that is starving in UMS since age.

**Comment by Nadahar** [comment]
> have a media with a video level > 4.2

Ah, I didn't know that it actually supported 4.2. We should really try to revive that old code that they didn't want in UMS, it would be a much better solution.

**Comment by Sami32** [comment]
@Nadahar Yes definitively. Not saying that it also solved H.264 parsing bugs.

File: src/main/external-resources/renderers/Sony-PlayStation3.conf [hunk 1]
```diff
@@ -20,33 +20,28 @@ RendererIcon = ps3.png
...
 MediaInfo = true
 
 # Supported video formats:
-Supported = f:mpegps     v:mpeg1|mpeg2|mp4|h264           a:ac3|lpcm           m:video/mpeg
-Supported = f:mpegts     v:mpeg2|mpeg4sp|h264             a:ac3|lpcm|mpa       m:video/mpeg
-Supported = f:avi|divx   v:divx|mjpeg|mp4|mpeg4sp         a:mp3|lpcm|mpa|ac3   m:video/x-divx     gmc:0
-Supported = f:mp4        v:mp4|h264|mpeg4sp               a:ac3                m:video/mp4
-Supported = f:mp4        v:mp4|h264                       a:aac-lc             m:video/mp4                n:2
-Supported = f:wmv        v:wmv|vc1                        a:wma                m:video/x-ms-wmv           n:2
+Supported = f:avi|divx   v:mpegasp|mpeg4sp|divx|mjpeg|xvid   a:mp3|lpcm|mpa|ac3       m:video/x-divx      gmc:0
```
Comment (MadokaAyukawa) [inline src/main/external-resources/renderers/Sony-PlayStation3.conf]: There needs to be a 4 in mpeg4asp.

**Comment by Nadahar** [comment]
Now the tests fail with:
```powershell
[ERROR]   FormatRecognitionTest.testPlaystationAudioMp3Compatibility:113 PS3 is incompatible with five channel MP3 expected:<false> but was:<true>
```
I don't know if the error is with the test or the configuration, but I doubt the PS3 can play five channel MP3. That said, for some reason this doesn't fail on macOS... which is even weirder.

**Comment by Sami32** [comment]
@Nadahar @MadokaAyuka Do you mind testing if these MP3 files work fine on your PS3? Thanks.
https://github.com/MediaArea/MediaInfo/files/793958/MP3_SURROUND.zip.txt
Remove the fake ".txt" extension before unzipping the MP3 samples.

**Comment by Sami32** [comment]
From USB or at least without transcoding

**Comment by Sami32** [comment]
@MadokaAyukawa If you don't mind iwould like you do a last test with this PS3 conf. before it get merged :wink:

Remove the `HalveBitrate = true` line and verify that transcoding still work without any stuttering for you. 

Thanks ;)

**Comment by MadokaAyukawa** [comment]
I've been watch to transcode multiple videos with those changes normally, no stuttering.

**Pull request merged**